src/pybind/mgr/cephadm/service_discovery.py

```diff
@@ -95,17 +87,18 @@ def _cp_dispatch(self, vpath: str) -> 'Root':
         cherrypy.request.path = ''
         return self
 
-    def __init__(self, mgr: "CephadmOrchestrator",
-                 port: int = 0,
-                 host: str = '',
-                 ssl_ca_cert: str = '',
-                 ssl_priv_key: str = ''):
+    def stop(self) -> None:
+        # we must call unsubscribe before stopping the server,
+        # otherwise the port is not released and we will get
+        # an exception when trying to restart it
+        self.unsubscribe()
+        super().stop()
+
+    def __init__(self, mgr: "CephadmOrchestrator", port: int = 0, host: str = '0.0.0.0'):
```

Comment:
technically the defaults are valid because the server will listen in `'0.0.0.0'` and a random port (if you pass 0) .. that what would happen if somebody tries to use the server without providing port/addr.